workflows/iuf/operations/vcs-update/vcs-update-working-branch.yaml

```diff
@@ -0,0 +1,188 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: vcs-update-working-branch-main-steps
+spec:
+  entrypoint: main
+  arguments:
+    parameters:
+      - name: product_manifest
+      - name: pristine_branch
+      - name: customer_branch
+  templates:
+    ### Main Steps ###
+    - name: main
+      steps:
+        - - name: get-vcs-secrets
+            template: get-vcs-secrets-template
+        - - name: gitea-update-content
+            template: gitea-update-content
+            arguments:
+              parameters:
+                - name: vcs_user_credentials_secret_name
+                  value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+        - - name: cleanup
+            template: cleanup-template
+            arguments:
+              parameters:
+                - name: vcs_user_credentials_secret_name
+                  value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+    ### Templates ###
+    ## get-vcs-secrets-template ##
+    - name: get-vcs-secrets-template
```

Comment:
You've used this in your previous VCS work. Is it possible to extract it into one place for modularity?

Reply:
I agree, I will file a Jira to track this.